AwsCryptographicMaterialProviders/dafny/AwsCryptographicMaterialProviders/src/Keyrings/AwsKms/AwsKmsHierarchicalKeyring.dfy

```diff
@@ -185,14 +185,18 @@ module AwsKmsHierarchicalKeyring {
         && var maybeSupplierModifies := if branchKeyIdSupplier.Some? then branchKeyIdSupplier.value.Modifies else {};
         && fresh(Modifies - keyStore.Modifies - cryptoPrimitives.Modifies - maybeSupplierModifies)
     {
-      var localCMC := new StormTracker.StormTracker(
-        entryCapacity := maxCacheSize as nat,
-        entryPruningTailSize := 1,
-        gracePeriod := gracePeriod.UnwrapOr(StormTracker.DefaultGracePeriod as Types.PositiveInteger) as Types.PositiveLong,
-                                            graceInterval := graceInterval.UnwrapOr(StormTracker.DefaultGraceInterval as Types.PositiveInteger) as Types.PositiveLong,
-                                                                                    fanOut := fanOut.UnwrapOr(StormTracker.DefaultFanOut as Types.PositiveInteger) as Types.PositiveLong,
-                                                                                                              inFlightTTL := inFlightTTL.UnwrapOr(StormTracker.DefaultInFlightTTL as Types.PositiveInteger) as Types.PositiveLong);
-      var cmc := new StormTrackingCMC.StormTrackingCMC(localCMC);
+      var cmc : Types.ICryptographicMaterialsCache;
+      if trackerSettings.None? || trackerSettings.value.gracePeriod > 0 {
+        var localCmc := new StormTracker.StormTracker(
+          entryCapacity := maxCacheSize as nat,
+          entryPruningTailSize := 1,
+          trackerSettings := trackerSettings
+        );
+        cmc := new StormTrackingCMC.StormTrackingCMC(localCmc);
+      } else {
+        var localCmc := new LocalCMC.LocalCMC(maxCacheSize as nat, 1);
+        cmc := new SynchronizedLocalCMC.SynchronizedLocalCMC(localCmc);
+      }
```

Comment:
I think moving this to the operation and then passing in the resultant `cmc` will let you reuse the `CreateCryptographicMaterialsCache` operation.